File: backend/Models/Payment.cs
using System.ComponentModel.DataAnnotations;

namespace MechanicApp.Server.Models
{
    public class Payment
    {
        public int Id { get; set; }

        public int? CustomerId { get; set; }

        [Range(0.01, 9999999.99, ErrorMessage = "Amount must be greater than zero.")]
        public decimal Amount { get; set; }

        [Required, StringLength(30)]
        public string PaymentMethod { get; set; } = "Cash";

        [Required(ErrorMessage = "The Reference # field is required, please fill the field to continue!")]
        [StringLength(100)]
        public string? ReferenceNumber { get; set; }

        public DateTime? PaymentDate { get; set; }

        [StringLength(500)]
        public string? Notes { get; set; }

        public int? CurrencyId { get; set; }

        public decimal? OriginalAmount { get; set; }

        public int? OriginalCurrencyId { get; set; }

        public DateTime? CreatedAt { get; set; }

        // Input-only: list of repair order IDs for multi-order payment
        [Required(ErrorMessage = "The Repair Order field is required, please fill the field to continue!")]
        [MinLength(1, ErrorMessage = "At least one Repair Order must be selected.")]
        public int[]? RepairOrderIds { get; set; }

        // Populated from JOINs (read-only display fields)
        public string? OrderInfo { get; set; }
        public string? CarInfo { get; set; }
        public string? CustomerName { get; set; }
        public decimal? OrderTotal { get; set; }
        public string? CurrencySymbol { get; set; }
        public string? OriginalCurrencySymbol { get; set; }
    }
}

File: frontend/src/app/models/payment.ts
export interface Payment {
  id?: number;
  repairOrderIds?: number[];
  customerId?: number;
  amount: number;
  paymentMethod: string;
  referenceNumber?: string;
  paymentDate?: string;
  notes?: string;
  currencyId?: number;
  originalAmount?: number;
  originalCurrencyId?: number;
  createdAt?: string;
  // Display fields from JOINs
  orderInfo?: string;
  carInfo?: string;
  customerName?: string;
  orderTotal?: number;
  currencySymbol?: string;
  originalCurrencySymbol?: string;
}
